src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -412,6 +412,111 @@ public static DynamoModel Start(IStartConfiguration configuration)
             return new DynamoModel(configuration);
         }
 
+        private static FileVersion FindPreviousVersion(string rootFolder, FileVersionInfo currentVersionInfo)
```

Comment:
This method needs rework, sir. The way to do this is:

1. Enumerate folders under `rootFolder` for a bunch of folder names (e.g. `0.7`, `0.8`, `0.9`, ...)
2. Break them down into multiple `FileVersion`, sort them (for this you need operator in `FileVersion`)
3. Write unit test for `FileVersion` operators :)